diff --git a/ports-juce5/tal-filter/source/TalCore.cpp b/ports-juce5/tal-filter/source/TalCore.cpp
--- a/ports-juce5/tal-filter/source/TalCore.cpp
+++ b/ports-juce5/tal-filter/source/TalCore.cpp
@@ -50,6 +50,15 @@ void TalCore::setParameter (int index, float newValue)
     {
         return;
     }
+    
+    // Scale parameters correctly for LV2
+    switch (index)
+    {
+        case FILTERTYPE:
+            newValue = newValue * 7.0f + 1.0f;
+            break;
+    }
+    
     params[index] = newValue;
     
     switch (index)
     {
+        case CUTOFF:
+            engine->setCutoff(newValue);
+            break;
         case RESONANCE:
+            engine->setResonance(newValue);
+            break;
         case FILTERTYPE:
+            engine->setFilterType((int)(newValue));
+            break;
         case LFORATE:
             engine->setSync((int)params[LFOSYNC], newValue);
             break;